optimize/backend/src/main/java/io/camunda/optimize/TomcatConfig.java

```diff
@@ -0,0 +1,260 @@
+    final FilterRegistrationBean<URLRedirectFilter> registration = new FilterRegistrationBean<>();
+    registration.addUrlPatterns("/*");
+    registration.setFilter(filter);
+    return registration;
+  }
+
+  @Bean
+  FilterRegistrationBean<ResponseHeadersFilter> responseHeadersInjector() {
+    LOG.debug("Registering filter 'responseHeadersInjector'...");
+    final ResponseHeadersFilter responseHeadersFilter =
+        new ResponseHeadersFilter(configurationService);
+    final FilterRegistrationBean<ResponseHeadersFilter> registrationBean =
+        new FilterRegistrationBean<>();
+    registrationBean.addUrlPatterns("/*");
+    registrationBean.setFilter(responseHeadersFilter);
+    return registrationBean;
+  }
+
+  public int getPort(final String portType) {
+    final String portProperty = environment.getProperty(portType);
+    if (portProperty != null) {
+      try {
+        return Integer.parseInt(portProperty);
+      } catch (final NumberFormatException exception) {
+        throw new OptimizeConfigurationException("Error while determining container port");
+      }
+    }
+
+    if (portType.equals(EnvironmentPropertiesConstants.HTTPS_PORT_KEY)) {
+      return configurationService.getContainerHttpsPort();
+    }
+
+    final Optional<Integer> httpPort = configurationService.getContainerHttpPort();
+    if (httpPort.isEmpty()) {
+      throw new OptimizeConfigurationException("HTTP port not configured");
+    }
+    return httpPort.get();
+  }
+
+  public Optional<String> getContextPath() {
+    // If the property is set by env var (the case when starting a new Optimize in ITs), this takes
+    // precedence over config
+    final Optional<String> contextPath = Optional.ofNullable(environment.getProperty(CONTEXT_PATH));
+    if (contextPath.isEmpty()) {
+      return configurationService.getContextPath();
+    }
+    return contextPath;
+  }
+
+  private SSLHostConfig getSslHostConfig() {
+    final SSLHostConfig sslHostConfig = new SSLHostConfig();
+    sslHostConfig.setHostName(configurationService.getContainerHost());
+
+    final SSLHostConfigCertificate cert =
+        new SSLHostConfigCertificate(sslHostConfig, Type.UNDEFINED);
+    cert.setCertificateKeystoreFile(configurationService.getContainerKeystoreLocation());
+    cert.setCertificateKeystorePassword(configurationService.getContainerKeystorePassword());
+    sslHostConfig.addCertificate(cert);
+
+    return sslHostConfig;
+  }
+
+  private void enableGzipSupport(final Connector connector) {
+    connector.setProperty("compression", "on");
+    connector.setProperty("compressionMinSize", "23");
+    connector.setProperty("compressionNoCompressionMethods", ""); // all methods
+    connector.setProperty("useSendfile", "false");
+    connector.setProperty("compressableMimeType", String.join(",", COMPRESSED_MIME_TYPES));
+  }
+
+  private void configureHttpConnector(final Connector connector) {
```

Comment:
Connector settings in this method and `configureHttpsConnector()` are doing the same thing in some cases. Could you use a `private` helper method for the ones the two methods have in common?